test/test_dynamic_shapes.py

```diff
@@ -0,0 +1,309 @@
+        z = x.narrow_copy(LAST_DIM, 0, x.sym_size(LAST_DIM) - 1)
+        self.assertEqual(int(z.sym_size(2)), 2)
+
+    @skipIfNoSympy
+    def test_symint_vargs(self):
+        shape_env = ShapeEnv()
+        x = create_symbolic_tensor("x", torch.randn(5, 4, 3), shape_env)
+        y = create_symbolic_tensor("y", torch.randn(1, 4, 1), shape_env)
+
+        # varargs
+        z = y.expand(x.sym_size(0), y.sym_size(1), x.sym_size(2))
+        self.assertEqual(int(z.sym_size(0)), 5)
+        self.assertEqual(int(z.sym_size(1)), 4)
+        self.assertEqual(int(z.sym_size(2)), 3)
+
+        # shape list
+        z = y.expand((x.sym_size(0), y.sym_size(1), x.sym_size(2)))
+        self.assertEqual(int(z.sym_size(0)), 5)
+        self.assertEqual(int(z.sym_size(1)), 4)
+        self.assertEqual(int(z.sym_size(2)), 3)
+
+        # mixed python symints and ints
+        z = y.expand(x.sym_size(0), y.sym_size(1), 3)
+        self.assertEqual(int(z.sym_size(0)), 5)
+        self.assertEqual(int(z.sym_size(1)), 4)
+        self.assertEqual(int(z.sym_size(2)), 3)
+
+        # mixed python symints and ints in a list
+        z = y.expand((x.sym_size(0), y.sym_size(1), 3))
+        self.assertEqual(int(z.sym_size(0)), 5)
+        self.assertEqual(int(z.sym_size(1)), 4)
+        self.assertEqual(int(z.sym_size(2)), 3)
+
+        # mixed python symints and ints
+        z = y.expand(5, y.sym_size(1), x.sym_size(2))
+        self.assertEqual(int(z.sym_size(0)), 5)
+        self.assertEqual(int(z.sym_size(1)), 4)
+        self.assertEqual(int(z.sym_size(2)), 3)
+
+        # mixed python ints and symints in a list
+        z = y.expand((5, y.sym_size(1), x.sym_size(2)))
+        self.assertEqual(int(z.sym_size(0)), 5)
+        self.assertEqual(int(z.sym_size(1)), 4)
+        self.assertEqual(int(z.sym_size(2)), 3)
+
+    @skipIfNoSympy
+    def test_size_expressions(self):
+        shape_env = ShapeEnv()
+        x = create_symbolic_tensor("x", torch.randn(5), shape_env)
+        expand_x = x.expand(x.sym_size(0), x.sym_size(0))
+        if expand_x.sym_size(0) > 3:
+            result = expand_x + expand_x
+        else:
+            result = expand_x + expand_x
+
+        gt_op = shape_env.guards[0][0]
+        self.assertTrue(isinstance(gt_op, sympy.core.relational.StrictGreaterThan))
+        self.assertTrue(str(x.sym_size(0)), str(gt_op.args[0]))
+        self.assertTrue(str(expand_x.sym_size(1)), str(x.sym_size(0)))
+        self.assertTrue(str(expand_x.sym_size(1)), str(result.sym_size(0)))
+
+    def test_fx_trace_intlist(self):
+        class CustomModule(torch.nn.Module):
+            def forward(self, x):
+                bs, c, h, w = x.shape
+                return F.pad(x, (0, w % 2, 0, h % 2, 0, 0))
+
+        m = CustomModule()
+        x = torch.rand(1, 3, 4, 4)
+        # should not TypeError: pad(): argument 'pad' (position 2) must be
+        # tuple of ints, not tuple
+        torch.fx.symbolic_trace(m)
```

Comment:
Technically, we do not need to create a custom module to repro the issue, just a function:
```python
def foo(x):
    bs, c, h, w = x.shape
    return F.pad(x, (0, w % 2, 0, h % 2, 0, 0))

torch.fx.symbolic_trace(foo)
```